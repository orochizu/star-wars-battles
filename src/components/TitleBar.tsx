import React, { useCallback, useState } from 'react';

import { useTranslation } from 'react-i18next';

import {
    AppBar,
    FormControlLabel,
    Grid,
    Switch,
    Toolbar,
    Typography,
} from '@material-ui/core';

import Link from './Link';

const TitleBar: React.FC = () => {
    const [isEnglish, setIsEnglish] = useState(true);
    const { t, i18n } = useTranslation();

    const toggleLanguage = useCallback(
        (value: boolean) => {
            setIsEnglish(value);
            i18n.changeLanguage(isEnglish ? 'pl' : 'en');
        },
        [i18n, isEnglish],
    );

    return (
        <AppBar position="static">
            <Toolbar>
                <Grid container justify="flex-start">
                    <Grid item xs="auto">
                        <Link to="/">
                            <Typography align="center">
                                {t('APP_TITLE')}
                            </Typography>
                        </Link>
                    </Grid>
                </Grid>
                <Grid container justify="flex-end">
                    <Grid item>
                        <FormControlLabel
                            control={
                                <Switch
                                    checked={isEnglish}
                                    onChange={event =>
                                        toggleLanguage(event.target.checked)
                                    }
                                />
                            }
                            label={t('SWITCH_LANGUAGE')}
                            labelPlacement="start"
                        />
                    </Grid>
                </Grid>
            </Toolbar>
        </AppBar>
    );
};

export default TitleBar;
