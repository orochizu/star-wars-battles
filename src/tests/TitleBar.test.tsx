import React from 'react';

import { createRenderer } from 'react-test-renderer/shallow';
import TitleBar from '../components/TitleBar';

jest.mock('react-i18next', () => {
    return {
        useTranslation: () => ({
            t: (key: string) => key.toUpperCase(),
        }),
    };
});

const component = <TitleBar />;

it('TitleBar.tsx should match snapshot', () => {
    const renderer = createRenderer();
    renderer.render(component);

    const snapshot = renderer.getRenderOutput();

    expect(snapshot).toMatchSnapshot();
});
