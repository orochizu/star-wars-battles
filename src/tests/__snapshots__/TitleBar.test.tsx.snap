// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar.tsx should match snapshot 1`] = `
<ForwardRef(WithStyles)
  position="static"
>
  <ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      container={true}
      justify="flex-start"
    >
      <ForwardRef(WithStyles)
        item={true}
        xs="auto"
      >
        <ForwardRef(render)
          to="/"
        >
          <ForwardRef(WithStyles)
            align="center"
          >
            APP_TITLE
          </ForwardRef(WithStyles)>
        </ForwardRef(render)>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      container={true}
      justify="flex-end"
    >
      <ForwardRef(WithStyles)
        item={true}
      >
        <ForwardRef(WithStyles)
          control={
            <ForwardRef(WithStyles)
              checked={true}
              onChange={[Function]}
            />
          }
          label="SWITCH_LANGUAGE"
          labelPlacement="start"
        />
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;
